build-system/babel-plugins/babel-plugin-transform-jss/index.js

```diff
@@ -44,9 +45,28 @@ module.exports = function ({types: t, template}) {
     return filename.endsWith('.jss.js');
   }
 
-  function compileJss(JSS) {
-    const jss = create();
-    jss.setup(preset());
+  const seen = new Set();
+  function compileJss(JSS, filename) {
+    const filehash = crypto
+      .createHash('sha256')
+      .update(filename)
+      .digest('base64')
+      .slice(0, 7);
+    const jss = create({
+      ...preset(),
+      createGenerateId: () => {
+        return (rule) => {
+          const className = `${rule.key}-${filehash}`;
```

Comment:
Two requests here:

1. Can we instead do `${filehash}-${rule.key}`? IMHO that'd be more natural.
2. Can we camel->dashcase the `rule.key`? E.g. `hideScroll` -> `hide-scroll`.

Reply:
Why is (1) more natural? My instinct when visually scanning for classnames is to prefer that the human-written part to be first.

re (2): sure!

Reply:
I thought it'd be more natural since the class name while looking at the code structure is `extensions/1.0/base-carousel/base-carousel.jss.js-button`. We could actually use this class name (with CSS escapes, of course) but it's too cumbersome. So we are contracting the first part to a hash.

However, I see your point. It'd be easier to scan/find by local classes first. Let's try your way first and see how it feels for devex.

Reply:
Sounds good, we can easily change the order in the future if when we want to 👍